src/System.Management.Automation/engine/CommandSearcher.cs

```diff
@@ -635,7 +635,41 @@ private CommandInfo GetInfoFromPath(string path)
                     break;
                 }
 
-                if (String.Equals(extension, StringLiterals.PowerShellScriptFileExtension, StringComparison.OrdinalIgnoreCase))
+                // handle case where we are called by the OS to handle a shebang script using us as the interpreter
+                bool isShebangScript = false;
+                try
+                {
+                    using(FileStream fileStream = new FileStream(path, FileMode.Open, FileAccess.Read))
+                    {
+                        byte[] bytes = new byte[2];
+                        int bytesRead = fileStream.Read(bytes, 0, 2);  // read the first two bytes to determine if shebang
+                        if (bytesRead == 2)
+                        {
+                            if (bytes[0] == '#' && bytes[1] == '!')
+                            {
+                                // see if we are supposed to be the interpreter
+                                using(StreamReader file = new StreamReader(fileStream))
+                                {
+                                    string interpreter = file.ReadLine();
+                                    System.Reflection.Assembly assembly = System.Reflection.Assembly.GetEntryAssembly();
```

Comment:
We should certainly cache this since this if using shebang with PowerShell becomes popular.  I have to explicitly check if the interpreter is exactly the same as the running PowerShell otherwise I let the OS handle it as we would want to support scripts that target specific versions of PowerShell instead of just the system default one.  Actually this reminds me that the current code won't work with

```
#!/usr/bin/env powershell
```